libs/ledger/src/chain/block_coordinator.cpp

```diff
@@ -505,38 +504,21 @@ BlockCoordinator::State BlockCoordinator::OnPreExecBlockValidation()
   if (!is_genesis)
   {
     BlockPtr previous = chain_.GetBlock(current_block_->previous_hash);
-    if (!previous)
-    {
-      FETCH_LOG_WARN(LOGGING_NAME, "Block validation failed: No previous block in chain (0x",
-                     current_block_->hash.ToHex(), ')');
-
-      RemoveBlock(current_block_->hash);
-      return State::RESET;
-    }
 
-    consensus_->UpdateCurrentBlock(*previous);  // Only update with valid blocks
     auto result = consensus_->ValidBlock(*current_block_);
 
     if (!(result == ConsensusInterface::Status::YES))
     {
-      FETCH_LOG_WARN(LOGGING_NAME, "Block validation failed: Consensus failed to verify block (0x",
-                     current_block_->hash.ToHex(), ')');
+      FETCH_LOG_ERROR(LOGGING_NAME,
+                      "Block validation failed: Block coordinator failed to verify block (0x",
+                      current_block_->hash.ToHex(), ')', ". This should not happen.");
 
       RemoveBlock(current_block_->hash);
+      Reset();
       return State::RESET;
     }
 
-    // Check: Ensure the block number is continuous
-    uint64_t const expected_block_number = previous->block_number + 1u;
-    if (expected_block_number != current_block_->block_number)
-    {
-      FETCH_LOG_WARN(LOGGING_NAME, "Block validation failed: Block number mismatch. Expected: ",
-                     expected_block_number, " Actual: ", current_block_->block_number, " (0x",
-                     current_block_->hash.ToHex(), ')');
-
-      RemoveBlock(current_block_->hash);
-      return State::RESET;
-    }
+    consensus_->UpdateCurrentBlock(*previous);
 
```

Comment:
No need to check anymore that previous is not null?

Reply:
I put this check in the ValidBlock check
```
  if (current.hash.size() != DIGEST_LENGTH_BYTES ||
      current.previous_hash.size() != DIGEST_LENGTH_BYTES)
  {
    return Status::NO;
  }
```